x-pack/legacy/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -83,7 +83,11 @@ function getUniqueLayerCounts(layerCountsList: ILayerTypeCount[], mapsCount: num
 }
 
 function getIndexPatternsWithGeoFieldCount(indexPatterns: IIndexPattern[]) {
-  const fieldLists = indexPatterns.map(indexPattern => JSON.parse(indexPattern.attributes.fields));
+  const fieldLists = indexPatterns.map(indexPattern =>
+    indexPattern.attributes && indexPattern.attributes.fields
+      ? JSON.parse(indexPattern.attributes.fields)
```

Comment:
Maybe a warning should be logged about malformed saved object so it can be cleaned up? Something like: "Index-pattern ${id} is malformed and could cause problems"?

Reply:
Might be out of scope for this fix. I did a quick search to see how others were handling missing `fields` to potentially copy their handling of logging. Looking through 30ish examples, no one really logs the issue. For what we're doing here, harvesting the number of patterns with geo data, I think just handling malformed `indexPatterns` should be enough